[issue #13777] Fix improper exception handling and some other basic coding issues

iggzq opened the pull request:
## What is the purpose of the change
FIX https://github.com/alibaba/nacos/issues/13777: Fix improper exception handling and other minor issues
## Brief changelog

- Fix the incorrect member variable name SPRING_JXM_ENABLED in the ``NacosBootstrap`` class. The correct name should be SPRING_JMX_ENABLED.
- Fix the log message in the request method of the ``GrpcRequestAcceptor`` class. The incorrect message "un registered" has been corrected to "unregistered".
- Refactor the isNamespaceExist method in the ``NamespaceOperationService`` class into two separate methods: namespaceExists and validateNamespaceNotExists. This resolves the issue of improper exception catching in other classes caused by misuse of the original method.
- Update the isNamespaceExist method in the ``NamespaceValidationRequestFilter`` class to adapt to the new logic for checking namespace existence.
- Update the checkNamespaceIdExist method in the ``NamespaceInnerHandler`` class to align with the new namespace existence checking logic.
- Modify test cases in ``NamespaceInnerHandlerTest`` and ``NamespaceValidationRequestFilterTest`` to reflect the updated namespace existence checking logic.

iggzq committed:
style(core): 优化日志信息和代码格式

- 修改"un registered"为"unregistered"，规范日志信息表述
- 修改错误的变量定义

iggzq committed:
refactor(namespace): 重构命名空间存在性检查逻辑

- NamespaceOperationService将检查命名空间是否存在方法抽离出来，存在返回true，不存在返回false，使代码更加清晰。
- 修正一些因此修改产生变化的代码逻辑

iggzq committed:
Merge branch 'alibaba:develop' into develop

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/alibaba/nacos?pullRequest=13782) <br/>All committers have signed the CLA.

iggzq committed:
refactor(namespace): 重构命名空间校验逻辑

- 优化了 NamespaceOperationService 类中的namespaceExists方法异常打印内容
- 改进了 NamespaceValidationRequestFilter 中的命名空间存在性检查逻辑，将原先逻辑删除，在filter方法里直接调用namespaceOperationService.namespaceExists方法即可

KomachiSion commented on core/src/main/java/com/alibaba/nacos/core/service/NamespaceOperationService.java:
```diff
@@ -41,33 +42,33 @@
 @Service
 public class NamespaceOperationService {
-    
+
```
Please dont change the indent.

iggzq replied:
Okay, the code indentation has been reverted to its original state.

iggzq committed:
style(core): 优化代码格式和命名规范

- 调整代码缩进和空格，与原来保持一致

iggzq committed:
style(core): 优化代码格式和命名规范

- 调整代码缩进和空格，与原来保持一致

wuyfee commented:
$\color{red}{FAILURE}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/17573755517)
 ✅ - docker:   success 
 ❌ - deploy (standalone & cluster & standalone_auth):   failure 
 ❌ - e2e-java-test (standalone & cluster & standalone_auth):   skipped 
 ❌ - e2e-go-test (standalone & cluster):   skipped 
 ❌ - e2e-cpp-test (standalone & cluster):   skipped 
 ❌ - e2e-csharp-test (standalone & cluster):   skipped 
 ❌ - e2e-nodejs-test (standalone & cluster):   skipped 
 ❌ - e2e-python-test (standalone & cluster):   skipped 
 ✅ - clean (standalone & cluster & standalone_auth):   success

iggzq committed:
Merge branch 'alibaba:develop' into develop

iggzq committed:
refactor and test(core): 优化命名空间存在性验证逻辑并增加单元测试
- 重构 namespaceExists 方法，try-catch包裹默认namespace判断逻辑
- 将原先在NamespaceValidationRequestFilterTest类中的在校验命名空间时报错测试，改到NamespaceOperationServiceTest，因为现在异常处理和NamespaceValidationRequest类无关，和NamespaceOperationService类有关
- 在改到NamespaceOperationServiceTest类中增加异常处理测试用例，验证数据库查询失败时的返回结果
- 在NamespaceOperationServiceTest类中添加默认命名空间和非默认命名空间存在性验证的测试用例

iggzq commented:
The Maven test error in GitHub Actions has been resolved. The main issue was that the code logic for verifying the existence of namespaces has changed, and the original test method could no longer meet the testing requirements.

KomachiSion commented:
@iggzq  unit test can't pass. Please fix.

wuyfee commented:
$\color{red}{FAILURE}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/17600511626)
 ✅ - docker:   success 
 ❌ - deploy (standalone & cluster & standalone_auth):   failure 
 ❌ - e2e-java-test (standalone & cluster & standalone_auth):   skipped 
 ❌ - e2e-go-test (standalone & cluster):   skipped 
 ❌ - e2e-cpp-test (standalone & cluster):   skipped 
 ❌ - e2e-csharp-test (standalone & cluster):   skipped 
 ❌ - e2e-nodejs-test (standalone & cluster):   skipped 
 ❌ - e2e-python-test (standalone & cluster):   skipped 
 ✅ - clean (standalone & cluster & standalone_auth):   success

iggzq committed:
Merge branch 'alibaba:develop' into develop

iggzq committed:
test:优化了测试用例，避免了不必要的异常处理

- 删除了 checkNamespaceIdExistWithException 测试方法。因为判断命名空间是否存在的逻辑已经发生改变，namespaceInnerHandler.checkNamespaceIdExist此方法无需再捕获NacosApiException异常来设置true或false，所以此测试方法可以直接删除

iggzq commented:
> @iggzq unit test can't pass. Please fix.

Now it should be completely resolved:
 1. The issue was that previously, the ``namespaceInnerHandler.checkNamespaceIdExist`` method relied on catching the thrown NacosApiException to indicate whether a namespace existed. That's why the test was added. Now, the ``namespaceOperationService.namespaceExists`` method directly catches the exception and returns a boolean value (true or false), making the original test obsolete and safe to remove. 
 2. Running ``mvn clean install -DskipITs`` locally passes all tests.

KomachiSion commented:
PR merged, @iggzq Would you mind provider an dingtalk number to invite you join Nacos contributor group.

iggzq commented:
> PR merged, @iggzq Would you mind provider an dingtalk number to invite you join Nacos contributor group.

ok，my dingtalk number is gaozongqi1024

wuyfee commented:
$\color{red}{FAILURE}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/17637167381)
 ✅ - docker:   success 
 ❌ - deploy (standalone & cluster & standalone_auth):   failure 
 ❌ - e2e-java-test (standalone & cluster & standalone_auth):   skipped 
 ❌ - e2e-go-test (standalone & cluster):   skipped 
 ❌ - e2e-cpp-test (standalone & cluster):   skipped 
 ❌ - e2e-csharp-test (standalone & cluster):   skipped 
 ❌ - e2e-nodejs-test (standalone & cluster):   skipped 
 ❌ - e2e-python-test (standalone & cluster):   skipped 
 ✅ - clean (standalone & cluster & standalone_auth):   success
